frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerManagerThread.java

```diff
@@ -0,0 +1,474 @@
+
+    public WorkerManagerState getState() {
+        return state;
+    }
+
+    public WorkerManagerLifeCycle getLifeCycle() {
+        return lifeCycle;
+    }
+
+    public WorkerManagerThread(
+            ConfigManager configManager,
+            EventLoopGroup backendEventGroup,
+            int port,
+            Model model,
+            BatchAggregator aggregator,
+            WorkerManagerStateListener listener) {
+
+        this.workerId = String.valueOf(port); // Unique across all workers.
+        this.configManager = configManager;
+        this.backendEventGroup = backendEventGroup;
+        this.port = port;
+        this.model = model;
+        this.aggregator = aggregator;
+        this.gpuId = gpuId;
+        this.listener = listener;
+        startTime = System.currentTimeMillis();
+        lifeCycle = new WorkerManagerLifeCycle(configManager, model);
+        replies = new ArrayBlockingQueue<>(1);
+        workerLoadTime =
+                new Metric(
+                        getWorkerName(),
+                        String.valueOf(System.currentTimeMillis()),
+                        "ms",
+                        ConfigManager.getInstance().getHostName(),
+                        new Dimension("Level", "Host"));
+    }
+
+    @Override
+    public void run() {
+        int responseTimeout = model.getResponseTimeout();
+        Thread thread = Thread.currentThread();
+        thread.setName(getWorkerName());
+        currentThread.set(thread);
+        BaseModelRequest req = null;
+        int status = HttpURLConnection.HTTP_INTERNAL_ERROR;
+
+        try {
+            connect();
+
+            while (isRunning()) {
+                req = aggregator.getCtrlRequest(workerId, state);
+
+                if (req == null) {
+                    try {
+                        Thread.sleep(500);
+                    } catch (InterruptedException e) {
+                        logger.info("Waiting to ctrl requests ..");
+                    }
+                    continue;
+                }
+
+                long wtStartTime = System.currentTimeMillis();
+                backendChannel.writeAndFlush(req).sync();
+
+                long begin = System.currentTimeMillis();
+                ModelWorkerResponse reply = replies.poll(responseTimeout, TimeUnit.SECONDS);
+
+                long duration = System.currentTimeMillis() - begin;
+                logger.info("Backend response time: {}", duration);
+
+                if (reply != null) {
+                    aggregator.sendResponse(reply);
+                } else {
+                    int val = model.incrFailedInfReqs();
+                    logger.error("Number or consecutive unsuccessful inference {}", val);
+                    throw new WorkerInitializationException(
+                            "Backend worker did not respond in given time");
+                }
+                switch (req.getCommand()) {
+                    case LOAD:
```

Comment:
What is WorkerManagerThread or WorkerManagerLifeCycle's action if load fails? At least, scaleUp should not be proceeded.

Reply:
This may be an existing bug. It looks like we dont respond to ERROR state of the worker in the right manner. Have fixed this with an explicit condition in in the control loop.